compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -288,43 +295,32 @@ pub(crate) fn to_llvm_features<'a>(sess: &Session, s: &'a str) -> Option<LLVMFea
-        // llvm/llvm-project#111598
-        ("wasm32" | "wasm64", "wide-arithmetic") if get_version() < (20, 0, 0) => None,
-        ("sparc", "leoncasa") => Some(LLVMFeature::new("hasleoncasa")),
+        ("wasm32" | "wasm64", "wide-arithmetic") if get_version().0 < 20 => None,
+        (
+            "x86",
+            "amx-avx512" | "amx-fp8" | "amx-movrs" | "amx-tf32" | "amx-transpose" | "movrs",
+        ) if get_version().0 < 20 => None,
+        /*
+            Perform relatively complex feature conversion.
+        */
+        // Before LLVM 20 those two features were packaged together as b16b16
+        ("aarch64", "sve-b16b16" | "sme-b16b16") if get_version().0 < 20 => {
+            Some(LLVMFeature::new("b16b16"))
+        }
         // In LLVM 19, there is no `v8plus` feature and `v9` means "SPARC-V9 instruction available and SPARC-V8+ ABI used".
         // https://github.com/llvm/llvm-project/blob/llvmorg-19.1.0/llvm/lib/Target/Sparc/MCTargetDesc/SparcELFObjectWriter.cpp#L27-L28
         // Before LLVM 19, there was no `v8plus` feature and `v9` means "SPARC-V9 instruction available".
         // https://github.com/llvm/llvm-project/blob/llvmorg-18.1.0/llvm/lib/Target/Sparc/MCTargetDesc/SparcELFObjectWriter.cpp#L26
         ("sparc", "v8plus") if get_version().0 == 19 => Some(LLVMFeature::new("v9")),
-        ("powerpc", "power8-crypto") => Some(LLVMFeature::new("crypto")),
-        // These new `amx` variants and `movrs` were introduced in LLVM20
-        ("x86", "amx-avx512" | "amx-fp8" | "amx-movrs" | "amx-tf32" | "amx-transpose")
-            if get_version().0 < 20 =>
-        {
-            None
-        }
-        ("x86", "movrs") if get_version().0 < 20 => None,
-        ("x86", "avx10.1") => Some(LLVMFeature::new("avx10.1-512")),
-        ("x86", "avx10.2") if get_version().0 < 20 => None,
-        ("x86", "avx10.2") if get_version().0 >= 20 => Some(LLVMFeature::new("avx10.2-512")),
-        ("x86", "apxf") => Some(LLVMFeature::with_dependencies(
-            "egpr",
-            smallvec![
-                TargetFeatureFoldStrength::Both("push2pop2"),
-                TargetFeatureFoldStrength::Both("ppx"),
-                TargetFeatureFoldStrength::Both("ndd"),
-                TargetFeatureFoldStrength::Both("ccmp"),
-                TargetFeatureFoldStrength::Both("cf"),
-                TargetFeatureFoldStrength::Both("nf"),
-                TargetFeatureFoldStrength::Both("zu"),
-            ],
+        // Enable the evex512 target feature if an avx512 target feature is enabled.
+        ("x86", s) if s.starts_with("avx512") => Some(LLVMFeature::with_dependencies(
+            s,
+            smallvec![TargetFeatureFoldStrength::EnableOnly("evex512")],
         )),
+        ("x86", "avx10.2") => (get_version().0 >= 20).then(|| LLVMFeature::new("avx10.2-512")),
+        /*
+            By default, use the Rust's target feature name as-is.
```

Comment:
"the Rust's"?

Reply:
Simple mistake.
Since I abandon the approach 2 on the commit 5 (based on your another feedback), this comment will be removed.